course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala

```diff
@@ -96,14 +100,19 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
     private def enrolUser(request: Request, courseId: String, batchId: String, userId: String): Unit = {
         val batchData: CourseBatch = courseBatchDao.readById(courseId, batchId, request.getRequestContext)
         val enrolmentData: UserCourses = userCoursesDao.read(request.getRequestContext, userId, courseId, batchId)
-        validateEnrolment(batchData, enrolmentData, true)
+        val id:util.List[String]=List(userId)
+        val ListOfUserDataForCourse:CourseUser=courseUserDao.read(request.getRequestContext,courseId,id)
```

Comment:
we should add this code in upsertEnrolment method

Reply:
done